parlai/agents/transformer/modules.py

```diff
@@ -580,7 +580,8 @@ def forward(self, tensor, mask):
         """
         Forward pass.
         """
-        tensor = tensor + self.dropout(self.attention(tensor, mask=mask))
+        attended_tensor, _ = self.attention(tensor, mask=mask)
```

Comment:
Hmm does it make sense to dropout separately each forward? I don’t think so. This will affect Unlikelihood stuff.

Let’s keep the dropout on the cache but leave a comment warning.

Reply:
We need to separately dropout each forward of the MHA, because each forward relies on a different uncache-able query, etc. What kind of comment warning did you have in mind in this case?